flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierTracker.java

```diff
@@ -230,8 +227,7 @@ public long getAlignmentDurationNanos() {
 	}
 
 	@Override
-	public void checkpointSizeLimitExceeded(long maxBufferedBytes) throws Exception {
-		throw new UnsupportedOperationException("This should never happened as this class doesn't block any data");
+	public void registerInputGate(CheckpointedInputGate inputGate) {
```

Comment:
we can avoid introducing the interface method and this empty implementation by https://github.com/apache/flink/pull/11351/files#r397602529